### PR TITLE
docs: Minor tweaks + use roxygen Markdown + bring dev mod auto.

### LINKED ISSUE #910: Remove vignettes/articles/nc after successful site build
> > TODO: eventually remove vignettes/articles/nc after successful build.

If this is just for gh-pages, I'm happy to make an issue for a followup PR later. `gh-pages` isn't in the main branch, so it can have many copies of files

_Originally posted by @schloerke and @olivroy in https://github.com/rstudio/leaflet/issues/903#issuecomment-2023894984_

### PR DESCRIPTION
**Pull request by olivroy**
Also fix #910.

Enabling roxygen markdown doc with `usethis::use_roxygen_md()` will help a little. I didn't run `roxygen2md::roxygen2md()` because the diff would have gotten too big.

I couldn't really find how to download directly.

    Error in ncdf4::nc_open(filename, readunlim = FALSE, suppress_dimvals = TRUE) : 
    Error in nc_open trying to open file https://rstudio.github.io/leaflet/nc/oisst-sst.nc (return_on_error= FALSE )

### CODE REVIEW

**Comment by schloerke** [comment]
> I didn't run roxygen2md::roxygen2md() because the diff would have gotten too big.

If this is an automated call, I'd be up for an isolated PR with a single commit of just the changes from the auto conversion. 

Could it be a follow up PR? Or does it need to happen before this one?

If it looks too crazy, we can back out... but I'm still curious.

**Comment by schloerke** [comment]
I am sorry about the windows install failures. I'll try to fix that tomorrow. Looks like it's trying to install tools 4.2 when it needs 4.3

**Comment by olivroy** [comment]
> > I didn't run roxygen2md::roxygen2md() because the diff would have gotten too big.
> 
> If this is an automated call, I'd be up for an isolated PR with a single commit of just the changes from the auto conversion.
> 
> Could it be a follow up PR? Or does it need to happen before this one?
> 
> If it looks too crazy, we can back out... but I'm still curious.

I can do follow-up ! It is mostly automated, but requires some clean-up afterwards. But never a good idea to mix opinionated changees with automated ones as they can get lost!

File: vignettes/articles/raster.Rmd [hunk 1]
```diff
@@ -53,7 +53,8 @@ library(leaflet)
 ```{r eval=FALSE}
 library(raster)
 library(leaflet)
-r <- raster("https://rstudio.github.io/leaflet/nc/oisst-sst.nc")
+# you can download by accessing https://rstudio.github.io/leaflet/nc/oisst-sst.nc
+r <- raster("nc/oisst-sst.nc")
```
[the remark this answers is not in the document]
Comment (schloerke) [reply vignettes/articles/raster.Rmd:57]: It does not work with urls. keeping as is

**Pull request merged**